dxp/demystifying_aui_input_tags.md

```diff
@@ -0,0 +1,727 @@
+				step="1"
+			/>
+			<div class="clay-range-track"></div>
+			<div class="clay-range-progress">
+				<div class="clay-range-thumb"></div>
+			</div>
+		</div>
+	</label>
+</div>
+```
+
+## Resource
+
+This `aui:input` type outputs a `readonly` text `input` with JavaScript that highlights all text inside on click. This type passes the attributes `id`, `title`, and `value` to the `liferay-ui:input-resource` taglib which outputs the `readonly` input. The taglib `liferay-ui:input-resource` always namespaces the `id`. The attribute `useNamespace="<%= false %>"` does nothing here.
+
+| Attributes | Description                                                                |
+| ---------- | -------------------------------------------------------------------------- |
+| `cssClass` | Adds classes to `<input class="form-control" />`                           |
+| `id`       | A `string` to use as the `id`. This is always namespaced.                  |
+| `name`     | Same as `id`. This attribute is not output by `liferay-ui:input-resource`. |
+| `title`    | A `string` to use in the `title` attribute.                                |
+| `value`    | A `url` that gets passed to the `liferay-ui:input-resource` taglib.        |
+
+```jsp
+<aui:input label="Cadetblue" name="cadetblue" type="resource" />
+```
+
+Output:
+
+```html
+<div class="form-group input-resource-wrapper">
+	<label class="control-label" for="_portlet_namespace_cadetblue">
+		Cadetblue
+	</label>
+	<input
+		class="form-control form-text lfr-input-resource "
+		id="_portlet_namespace_cadetblue"
+		onclick="this.select();"
+		readonly="true"
+		type="text"
+		value="null"
+	/>
+</div>
+```
+
+## Time Zone
+
+The `aui:input` type `timeZone` outputs a `select` element containing time zones from around the world. The attributes in the `aui:input` get passed to the taglib `liferay-ui:input-time-zone` which outputs a `<select>` element with `<options>`.
+
+| Attributes     | Description                                                                                                                    |
+| -------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| `cssClass`     | Adds classes to `<select class="form-control" />`.                                                                             |
+| `daylight`     | TODO: Any `string`                                                                                                             |
+| `disabled`     | A `boolean` value that enables or disables the `input` (e.g., `disabled="<%= true %>"`)                                        |
+| `displayStyle` | TODO                                                                                                                           |
+| `id`           | Not supported, this uses the `name` attribute to generate the `id`.                                                            |
```

Comment:
The comma here creates a run-on sentence. Also the "this" is confusing; I gather you mean the tag, but it could also be interpreted as the attribute. So maybe this could be:

> Not supported (the tag uses the `name` attribute to generate the `id`).